server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java

```diff
@@ -0,0 +1,133 @@
+        checkPython();
+        deploymentScript.append(String.format("python %s %s", PYTHON_CONFIGURE_SQL_FILENAME, relativePathToSQLConfigureFile));
+    }
+
+    /**
+     will add a python script which executes a file on the warden container
+
+     @param appName               name of the created app
+     @param pathToFileOnContainer must be the path on the warden container. Probably something like that:
+     "/home/vcap/app..."
+     */
+    public void executeFile(String appName, String pathToFileOnContainer) throws IOException {
+        copyFile(PYTHON_EXECUTE_FILENAME, PYTHON_EXECUTE_SOURCE);
+        checkPython();
+        deploymentScript.append(String.format("python %s %s %s", PYTHON_EXECUTE_FILENAME, appName, pathToFileOnContainer));
+    }
+
+    /**
+     will add a python script which read out the credentials of a service and write it to environment variables of the
+     container
+
+     @param appName     name of the created app
+     @param serviceName must be the service name of the provider not the service instance name
+     @param serviceType e.g. "mysql" for a mysql service
+     */
+    public void readCredentials(String appName, String serviceName, ServiceTypes serviceType) throws IOException {
+        copyFile(PYTHON_READ_CREDENTIALS_FILENAME, PYTHON_READ_CREDENTIALS_SOURCE);
+        checkPython();
+        deploymentScript.append(String.format("python %s %s %s %s", PYTHON_READ_CREDENTIALS_FILENAME, appName, serviceName, serviceType.getName()));
+    }
+
+    /**
+     will add a python script which replaces all occurrence of a string inside a file.
+
+     @param pathToLocalFile must be the relative path started from the output/scripts folder to the file
+     @param findStr         String which will be replaced
+     @param replaceStr      String which replaces the findStr
+     */
+    public void replaceStrings(String pathToLocalFile, String findStr, String replaceStr) throws IOException {
+        copyFile(PYTHON_REPLACE_STRINGS_FILENAME, PYTHON_REPLACE_STRINGS_SOURCE);
+        checkPython();
+        deploymentScript.append(String.format("python %s %s %s %s", PYTHON_REPLACE_STRINGS_FILENAME, pathToLocalFile, findStr, replaceStr));
+    }
+
+    private void copyFile(String fileName, String source) throws IOException {
+        if (!isAlreadyCopied(fileName)) {
+            InputStream inputStream = deploymentClass.getResourceAsStream(source + fileName);
+            String contentFile = IOUtils.toString(inputStream);
+            inputStream.close();
+            fileAccess.access(PYTHON_SCRIPTS_TARGET + fileName).appendln(contentFile).close();
+        }
+    }
+
+    private boolean isAlreadyCopied(String fileName) {
+        try {
+            fileAccess.getAbsolutePath(PYTHON_SCRIPTS_TARGET + fileName);
+        } catch (FileNotFoundException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private void checkPython() throws IOException {
```

Comment:
You do this `checkPyhton()` after every call to `copyFile()` so you can just include it that method

Reply:
Done